examples/resources/grafana_slo/resource_complex.tf

```diff
@@ -0,0 +1,54 @@
+resource "grafana_slo" "test" {
+  name        = "Complex Resource - Terraform Testing"
+  description = "Complex Resource - Terraform Description"
+  query {
+    freeformquery = "sum(rate(apiserver_request_total{code!=\"500\"}[$__rate_interval])) / sum(rate(apiserver_request_total[$__rate_interval]))"
+    type          = "freeform"
+  }
+  objectives {
+    value  = 0.995
+    window = "30d"
+  }
+  labels {
+    key   = "slokey"
+    value = "slokey"
+  }
+  alerting {
+    name = "alertingname"
+
+    labels {
+      key   = "alertingkey"
+      value = "alertingvalue"
+    }
+
+    fastburn {
+      annotations {
+        key   = "name"
+        value = "Critical - SLO Burn Rate Alert"
+      }
+      annotations {
+        key   = "description"
+        value = "Error Budget is burning at a rate greater than 14.4x."
+      }
+      labels {
+        key   = "type"
+        value = "slo"
+      }
+    }
+
+    slowburn {
+      annotations {
```

Comment:
This is the HCL way of doing a list / array? Is there a more "list-y" approach? My first thought here was that the second `annotations` field would override the first. If it makes sense to keep this structure (a "struct" per-annotation), then maybe naming them `annotation` makes sense:
```terraform

    slowburn {
      annotation {
        key   = "name"
        value = "Warning - SLO Burn Rate Alert"
      }
      annotation {
        key   = "description"
        value = "Error Budget is burning at a rate greater than 1x."
      }
```

Reply:
Yes - that is the HCL way of doing a list/array. I agree - it does seem like the `second` field would override the first, but it is how it's done.

I'll rename the fields to `annotation` and `label` so it's clearer.